pkg/kwokctl/cmd/create/cluster/cluster.go

```diff
@@ -79,6 +79,7 @@ func NewCommand(ctx context.Context) *cobra.Command {
 	cmd.Flags().StringVar(&flags.Options.KwokControllerImage, "kwok-controller-image", flags.Options.KwokControllerImage, `Image of kwok-controller, only for docker/nerdctl/kind runtime
 '${KWOK_IMAGE_PREFIX}/kwok:${KWOK_VERSION}'
 `)
+	cmd.Flags().Uint32Var(&flags.Options.KwokControllerPort, "kwok-controller-port", flags.Options.KwokControllerPort, `Port of kwok-controller, only for binary runtime`)
```

Comment:
The port set here should be the port that is eventually given to the host, the 10247 port in the container.

If this is not 0, 10247 in the container needs to be mapped to this port on the host.


binary: kwok-controller-port
container(docker/kind): kwok-controller-port -> 10247

Reply:
It is not relevant to the current feature, I think it is better to reduce the size of this PR. 
`kwok-controller-port` preferably in a separate PR, and then as an example for adding other ports.

Reply:
I agree that I'm including too many functionalities here, but I wanted to control the port binary runtime is going to expose, so I can have tests hitting the same port on every runtime...
Binary would use `net.GetUnusedPort`, which becomes 32763 or similar any time I run it. For other runtimes, it would be 10247. I can't specify port_forward.yaml file that is universal for all runtimes without this (or at least I am not sure how to do it). 
Do you have any suggestions to help me with this :relaxed: ?

Reply:
Maybe we can add net.PortAvailable for example and try setting controller port to 10247 on binary. If not used, use 10247. Otherwise, default to GetUnusedPort That should not break current behavior I think

Reply:
I think that's a tough one. Maybe we add `--controller-port` first, and then rebase this PR.

Reply:
I created a issue to track #261

Reply:
Sounds good to me :smile: On it!